File: orientado_objetos/sequencial/Fib_3.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package sequencial;

/**
 *
 * @author Jardel Carvalho
 */
public class Fib_3 {
    
    public static long fib(int n) { 
        long anterior, atual, guardaAtual;
        
        anterior = 0;
        atual = 1;
        
        for(; n != 1; n--) {
            guardaAtual = atual;
            atual = atual + anterior;
            anterior = guardaAtual;
        }
        
        return atual; 
    }
}

File: imperativo/sequencial/fib_3.c
long long int fib_3(int n) {
    long long int anterior = 0, atual = 1, guardaAtual;

    loop:
        guardaAtual = atual;
        atual = atual + anterior;
        anterior = guardaAtual;

        if(--n >= 1) 
            goto loop;
}   